specifyweb/frontend/js_src/lib/components/Molecules/FilePicker.tsx

```diff
@@ -137,26 +137,34 @@ export function FilePicker({
  */
 export const downloadFile = async (
   fileName: string,
-  text: string
+  data: string | Blob,
 ): Promise<void> =>
   new Promise((resolve) => {
     let fileDownloaded = false;
     const iframe = document.createElement('iframe');
     iframe.classList.add('absolute', 'hidden');
     iframe.addEventListener('load', () => {
       if (iframe.contentWindow === null || fileDownloaded) return;
+      let url;
       const element = iframe.contentWindow.document.createElement('a');
-      element.setAttribute(
-        'href',
-        `data:text/plain;charset=utf-8,${encodeURIComponent(text)}`
-      );
-      element.setAttribute('download', fileName);
+      if (typeof data === 'string') {
+        element.setAttribute(
+          'href',
+          `data:text/plain;charset=utf-8,${encodeURIComponent(data as string)}`
+        );
+        element.setAttribute('download', fileName);
+      } else {
+        url = window.URL.createObjectURL(data);
```

Comment:
nice! didn't know this was a thing

```suggestion
        url = URL.createObjectURL(data);
```